### PR TITLE
[MAINTENANCE] Delete legacy checkpoint

### PR DESCRIPTION
**Pull request by cdkini**
We've reached a point with the V1 checkpoint that it makes the legacy checkpoint fully obsolete. This PR is meant to delete that legacy class and any related components/behavior.

These include:
- `Checkpoint`
- `ActionListValidationOperator` (and `ValidationOperator` as a concept)
- `StoreValidationResultAction`
- `CheckpointConfigurator`
- `DataContext` CRUD (in favor of factory methods)
- Validation result based `ValidationAction` API

There still needs to be clean up after this PR:
- Rewrite of any coverage we may have lost (specifically around ID/PK, data docs, and various backends)
- Removal of additional classes like `CheckpointConfig`
- Promotion of V1 methods to appropriate namespaces (i.e. `from checkpoint.v1_checkpoint import Checkpoint`)

I'd ideally want smaller PRs to complete this effort but it feels easier to do it in one go to prevent having to write intermediate glue code to get CI passing.

- [x] Description of PR changes above includes a link to [an existing GitHub issue](https://github.com/great-expectations/great_expectations/issues)
- [x] PR title is prefixed with one of: [BUGFIX], [FEATURE], [DOCS], [MAINTENANCE], [CONTRIB]
- [x] Code is linted - run `invoke lint` (uses `ruff format` + `ruff check`)
- [x] Appropriate tests and docs have been updated

For more information about contributing, see [Contribute](https://docs.greatexpectations.io/docs/contributing/contributing_checklist).

After you submit your PR, keep the page open and **monitor the statuses of the various checks made by our continuous integration process at the bottom of the page. Please fix any issues that come up** and [reach out on Slack](https://greatexpectations.io/slack) if you need help. Thanks for contributing!

### CODE REVIEW

File: tests/validator/test_validator.py [hunk 1]
```diff
@@ -201,68 +201,6 @@ def test_validator_default_expectation_args__sql(
         )
 
 
-@pytest.mark.big
-def test_columns(
```
Comment (cdkini) [inline tests/validator/test_validator.py:205]: These are methods that exist on the legacy validator and aren't part of the new one - deleting because the test setup required some complex legacy checkpoints

File: tests/test_fixtures/great_expectations_v013_titanic.yml [hunk 2]
```diff
@@ -43,28 +43,6 @@ stores:
       class_name: TupleFilesystemStoreBackend
       base_directory: checkpoints/
 
-validation_operators:
```
Comment (cdkini) [inline tests/test_fixtures/great_expectations_v013_titanic.yml:46]: The only validation operator that existed was the `ActionListValidationOperator` - now that we've removed the legacy checkpoint, we can remove this entire module. This includes removing any references within YAML

File: tests/render/test_default_markdown_view.py [hunk 3]
```diff
@@ -133,340 +129,6 @@ def test_render_section_page():
...
------------------------------------------------------------
-Powered by [Great Expectations](https://greatexpectations.io/)
-""".replace(" ", "")  # noqa: E501
-        .replace("\t", "")
-        .replace("\n", "")
-    )
-
-
-def test_render_section_page_with_fixture_data_multiple_validations(
-    validation_operator_result,
```
Comment (cdkini) [inline tests/render/test_default_markdown_view.py:303]: No more validation operators means no more validation operator result so this test is obsolete

File: tests/render/test_default_markdown_view.py [hunk 4]
```diff
@@ -133,340 +129,6 @@ def test_render_section_page():
     )
 
 
-def test_snapshot_render_section_page_with_fixture_data(validation_operator_result):
```
Comment (cdkini) [inline tests/render/test_default_markdown_view.py:136]: No more validation operators means no more validation operator result so this test is obsolete

File: great_expectations/checkpoint/actions.py [hunk 5]
```diff
@@ -125,63 +121,6 @@ def _using_cloud_context(self) -> bool:
 
         return project_manager.is_using_cloud()
 
-    @public_api
-    def run(
```
Comment (cdkini) [inline great_expectations/checkpoint/actions.py:129]: All `run` and `_run` methods can be deleted on actions - we will promote `v1_run` to `run` in a follow-up

File: great_expectations/checkpoint/util.py [hunk 6]
```diff
@@ -179,231 +166,6 @@ def send_email(  # noqa: C901, PLR0913
         return "success"
 
 
-def get_substituted_validation_dict(
```
Comment (cdkini) [inline great_expectations/checkpoint/util.py:182]: All of the deleted utils have to do with legacy checkpoints - the remainder of the functions here are for actions and should be moved into that file

File: great_expectations/checkpoint/v1_checkpoint.py [hunk 7]
```diff
@@ -31,6 +31,7 @@
     )
 
 
+@public_api
```
Comment (cdkini) [inline great_expectations/checkpoint/v1_checkpoint.py:34]: Necessary to keep docs functioning

File: great_expectations/data_context/data_context/cloud_data_context.py [hunk 8]
```diff
@@ -798,75 +791,6 @@ def _validate_suite_unique_constaints_before_save(self, key: GXCloudIdentifier)
                 "expectation_suite, set overwrite_existing=True."
             )
 
-    @override
-    def add_checkpoint(  # noqa: PLR0913
```
Comment (cdkini) [inline great_expectations/data_context/data_context/cloud_data_context.py:802]: Since all checkpoint CRUD is now done through the "factory" and these methods utilize the legacy component, we should remove them for all contexts

File: great_expectations/data_context/migrator/configuration_bundle.py [hunk 9]
```diff
@@ -19,8 +19,6 @@
     DataContextVariables,  # noqa: TCH001
 )
 from great_expectations.data_context.types.base import (
-    CheckpointConfig,
```
Comment (cdkini) [inline great_expectations/data_context/migrator/configuration_bundle.py:22]: All this migrator code is to be deleted but ran into some cyclic import issues when trying to do so (cc @billdirks). Making minimal changes to get things working

File: great_expectations/data_context/store/checkpoint_store.py [hunk 10]
```diff
@@ -4,236 +4,25 @@
...
-
-        config = self.get(key=key)
-
-        return config
-
-
-# NOTE: Will eventually be promoted to 'CheckpointStore' once the legacy Checkpoint
-#       and its related classes are removed
-class V1CheckpointStore(Store):
+class CheckpointStore(Store):
```
Comment (cdkini) [inline great_expectations/data_context/store/checkpoint_store.py:25]: Promoted `V1CheckpointStore` to `CheckpointStore`

File: great_expectations/data_context/types/base.py [hunk 11]
```diff
@@ -2568,7 +2551,7 @@ def __init__(  # noqa: PLR0913
         name: Optional[str] = None,
         expectation_suite_name: Optional[str] = None,
         batch_request: Optional[dict] = None,
-        action_list: Optional[Sequence[ActionDict]] = None,
+        action_list: Optional[Sequence[dict]] = None,
```
Comment (cdkini) [inline great_expectations/data_context/types/base.py:2554]: Removing `CheckpointConfig` is a separate effort so just keeping this untyped for now

File: tests/actions/test_core_actions.py [hunk 12]
```diff
@@ -91,538 +85,6 @@ def __init__(self, status_code):
         self.content = json.dumps({"ok": "True"})
 
 
-@pytest.mark.big
```
Comment (cdkini) [inline tests/actions/test_core_actions.py:94]: Delete all tests except `v1_run` ones

File: tests/data_context/test_data_context.py [hunk 13]
```diff
@@ -1339,48 +1171,6 @@ def test_add_expectation_to_expectation_suite(empty_data_context_stats_enabled):
...
-        "runtime_parameters": {"query": "select * from titanic"},
-    }
-    validator = context.get_validator(
-        batch_request=RuntimeBatchRequest(**batch_request),
-        expectation_suite_name=suite_name,
-    )
-    validator.expect_table_row_count_to_equal(
-        value={
-            # unnecessarily complex URN which should resolve to the actual row count.
-            "$PARAMETER": "abs(-urn:great_expectations:stores:my_query_store:col_count - urn:great_expectations:stores:my_query_store:dist_col_count) + 4"  # noqa: E501
```
Comment (cdkini) [inline tests/data_context/test_data_context.py:1362]: No longer supporting URN so deleting test instead of rewriting

File: tests/data_context/store/test_v1_checkpoint_store.py [hunk 14]
```diff
@@ -15,7 +15,7 @@
 from great_expectations.core.validation_definition import ValidationDefinition
 from great_expectations.data_context import AbstractDataContext
 from great_expectations.data_context.cloud_constants import GXCloudRESTResource
-from great_expectations.data_context.store.checkpoint_store import V1CheckpointStore
+from great_expectations.data_context.store.checkpoint_store import CheckpointStore
```
Comment (cdkini) [inline tests/data_context/store/test_v1_checkpoint_store.py:18]: File will be renamed to just be `test_checkpoint_store`

File: tests/conftest.py [hunk 15]
```diff
@@ -1090,262 +1085,6 @@ def titanic_v013_multi_datasource_multi_execution_engine_data_context_with_check
     return context
 
 
-@pytest.fixture
-def titanic_pandas_data_context_with_v013_datasource_stats_enabled_with_checkpoints_v1_with_templates(  # noqa: E501
```
Comment (cdkini) [inline tests/conftest.py:1094]: We don't care for templating with new checkpoints so this whole fixture is outdated

File: great_expectations/data_context/data_context/abstract_data_context.py [hunk 16]
```diff
@@ -1061,15 +1021,6 @@ def list_active_stores(self) -> List[Store]:
             if store.get("name") in active_store_names  # type: ignore[arg-type,operator]
         ]
 
-    @public_api
-    def list_checkpoints(self) -> Union[List[str], List[ConfigurationIdentifier]]:
```
Comment (cdkini) [inline great_expectations/data_context/data_context/abstract_data_context.py:1065]: All the old CRUD has gotta go 👋🏽

File: docs/docusaurus/docs/snippets/checkpoints.py [hunk 17]
```diff
@@ -1,42 +0,0 @@
-# <snippet name="docs/docusaurus/docs/snippets/checkpoints.py setup">
-import great_expectations as gx
-from great_expectations.datasource.fluent import DataAsset, Datasource
-
-context = gx.get_context()
-# </snippet>
```
Comment (cdkini) [inline docs/docusaurus/docs/snippets/checkpoints.py:6]: This whole flow feels irrelevant to V1 so I'm deleting

File: docs/docusaurus/docs/snippets/checkpoints_and_actions.py [hunk 18]
```diff
@@ -1,469 +0,0 @@
-import os
-
-import great_expectations as gx
-from great_expectations.core.expectation_validation_result import (
-    ExpectationSuiteValidationResult,
```
Comment (cdkini) [inline docs/docusaurus/docs/snippets/checkpoints_and_actions.py:5]: Checkpoints and actions follow very different patterns in V1 so deleting this file

File: docs/docusaurus/docs/snippets/databricks_deployment_patterns_dataframe_python_configs.py [hunk 19]
```diff
@@ -1,98 +0,0 @@
-# isort:skip_file
-import pathlib
-
```
Comment (cdkini) [inline docs/docusaurus/docs/snippets/databricks_deployment_patterns_dataframe_python_configs.py:3]: REWRITE - we may want to write a new version of this test

File: docs/docusaurus/docs/snippets/databricks_deployment_patterns_dataframe_yaml_configs.py [hunk 20]
```diff
@@ -1,238 +0,0 @@
-# ruff: noqa: I001, PTH118, PTH109, PTH120
-import os
```
Comment (cdkini) [inline docs/docusaurus/docs/snippets/databricks_deployment_patterns_dataframe_yaml_configs.py:2]: Same as the other file but using YAML config - we can delete

File: docs/docusaurus/docs/snippets/how_to_create_a_new_checkpoint.py [hunk 21]
```diff
@@ -1,65 +0,0 @@
-import pathlib
-
-import great_expectations as gx
```
Comment (cdkini) [inline docs/docusaurus/docs/snippets/how_to_create_a_new_checkpoint.py:3]: REWRITE - we may want to write a new version of this test

File: docs/docusaurus/docs/snippets/postgres_deployment_patterns.py [hunk 22]
```diff
@@ -1,95 +0,0 @@
-# <snippet name="docs/docusaurus/docs/snippets/postgres_deployment_patterns.py imports">
-import great_expectations as gx
-from great_expectations.checkpoint import Checkpoint
-
-# </snippet>
```
Comment (cdkini) [inline docs/docusaurus/docs/snippets/postgres_deployment_patterns.py:5]: REWRITE - we may want to write a new version of this test

File: docs/docusaurus/docs/snippets/result_format.py [hunk 23]
```diff
@@ -1,16 +1,6 @@
-from typing import Any, Dict, List
-
 import pandas as pd
 
 import great_expectations as gx
-from great_expectations.checkpoint import Checkpoint
```
Comment (cdkini) [inline docs/docusaurus/docs/snippets/result_format.py:6]: REWRITE - we may want to write a new version of this test. ID/PK integration with V1 Checkpoints should be discussed and appropriately tested

File: tests/checkpoint/test_checkpoint.py [hunk 24]
```diff
@@ -1,1962 +0,0 @@
-from __future__ import annotations
-
-import logging
```
Comment (cdkini) [inline tests/checkpoint/test_checkpoint.py:3]: Reviewers, please take a look at these tests and make sure they're okay to delete. I'm primarily focused on keeping E2E/integration tests alive (and those are covered in other files)

File: tests/checkpoint/test_checkpoint_result_format.py [hunk 25]
```diff
@@ -1,3445 +0,0 @@
-from __future__ import annotations
-
-import logging
```
Comment (cdkini) [inline tests/checkpoint/test_checkpoint_result_format.py:3]: REWRITE - these tests are around ID/PK and should be rewritten if we want to keep this feature for V1

File: tests/checkpoint/test_checkpoint_with_fluent_datasources.py [hunk 26]
```diff
@@ -1,1639 +0,0 @@
-from __future__ import annotations
-
-import copy
-import logging
-import pickle
-from typing import TYPE_CHECKING, Optional
```
Comment (cdkini) [inline tests/checkpoint/test_checkpoint_with_fluent_datasources.py:6]: REWRITE - integrations around Spark, SQL, and Pandas should be rewritten (but I think they are covered in other files)

File: tests/data_context/migrator/conftest.py [hunk 27]
```diff
@@ -1,283 +0,0 @@
-from typing import Dict, Final, List, Optional, Tuple, Union
```
Comment (cdkini) [inline tests/data_context/migrator/conftest.py:1]: Rather than deleting the migrator (which caused import cycles), I'm just deleting tests so we can move forward without issue. Same comment applies to everything in this directory

File: tests/data_context/store/test_checkpoint_store.py [hunk 28]
```diff
@@ -1,465 +0,0 @@
-from __future__ import annotations
-
-import logging
```
Comment (cdkini) [inline tests/data_context/store/test_checkpoint_store.py:3]: Legacy checkpoint store is deleted so we don't need this anymore

File: tests/datasource/fluent/data_asset/data_connector/test_s3_data_connector.py [hunk 29]
```diff
@@ -751,150 +746,3 @@ def check_sameness(prefix, expected_output):
     check_sameness("a.x/b/c", "a.x/b/c/")
     check_sameness("path/to/folder.something/", "path/to/folder.something/")
     check_sameness("path/to/folder.something", "path/to/folder.something")
-
-
-@pytest.mark.aws_deps
-@mock_s3
-def test_s3_checkpoint_run_using_different_store_prefixes_successfully(
```
Comment (cdkini) [inline tests/datasource/fluent/data_asset/data_connector/test_s3_data_connector.py:758]: REWRITE - uses checkpoint to test stores so we should mock or find another way to test

File: tests/datasource/fluent/integration/integration_test_utils.py [hunk 30]
```diff
@@ -32,12 +32,12 @@
 logger = logging.getLogger(__name__)
 
 
-def run_checkpoint_and_data_doc(
+def run_checkpoint_and_data_doc(  # noqa: PLR0915
```
Comment (cdkini) [inline tests/datasource/fluent/integration/integration_test_utils.py:35]: Rewrote this method but I think the related test is xfailed

File: tests/datasource/fluent/integration/test_integration_datasource.py [hunk 31]
```diff
@@ -59,6 +57,7 @@
 
 
 # This is marked by the various backend used in testing in the datasource_test_data fixture.
+@pytest.mark.xfail(reason="This test has not yet been rewritten to use V1 checkpoints")
```
Comment (cdkini) [inline tests/datasource/fluent/integration/test_integration_datasource.py:60]: REWRITE - need to ensure integration with data docs is working

File: tests/datasource/fluent/integration/test_sql_datasources.py [hunk 32]
```diff
@@ -492,78 +483,6 @@ def test_sqlite(
 
         sqlite_ds.add_table_asset(asset_name, table_name=table_name)
 
-    @pytest.mark.parametrize(
```
Comment (cdkini) [inline tests/datasource/fluent/integration/test_sql_datasources.py:495]: REWRITE - need to ensure functionality with SQL datasources

File: tests/datasource/fluent/test_viral_snippets.py [hunk 33]
```diff
@@ -221,64 +221,6 @@ def test_ctx_delete_removes_datasource_from_yaml(
...
-    partitioner = PartitionerYearAndMonth(column_name="pickup_datetime")
-
-    batch_request = datasource.get_asset(asset_name).build_batch_request(
-        options={"year": year, "month": month}, partitioner=partitioner
-    )
-
-    validator = context.get_validator(batch_request=batch_request)
-    validator.save_expectation_suite()
-
-    checkpoint = context.add_checkpoint(name="my_checkpoint", validator=validator)
```
Comment (cdkini) [inline tests/datasource/fluent/test_viral_snippets.py:249]: This is no longer a pattern we use - I think we can delete

File: tests/execution_engine/test_sqlalchemy_execution_engine.py [hunk 34]
```diff
@@ -173,66 +171,6 @@ def test_instantiation_via_fluent_data_sources_with_kwargs(
     assert execution_engine.engine
 
 
-@pytest.mark.trino
-def test_instantiation_via_fluent_data_source__trino_add_sql(sa, empty_data_context):
```
Comment (cdkini) [inline tests/execution_engine/test_sqlalchemy_execution_engine.py:177]: I think we can leverage the Trino test in `tests/datasource/fluent/integration/test_sql_datasources.py` instead of this

File: docs/docusaurus/docs/snippets/postgres_deployment_patterns.py (hunk 22 above)
Comment (tyler-hoffman) [reply docs/docusaurus/docs/snippets/postgres_deployment_patterns.py:5]: This one should be covered by https://github.com/great-expectations/great_expectations/pull/9788

File: great_expectations/data_context/store/checkpoint_store.py [hunk 35]
```diff
@@ -306,3 +95,14 @@ def _update(self, key: DataContextKey, value: V1Checkpoint, **kwargs):
         except gx_exceptions.StoreBackendError as e:
             name = key.to_tuple()[0]
             raise ValueError(f"Could not update Checkpoint '{name}'") from e  # noqa: TRY003
+
+    @staticmethod
+    def default_checkpoints_exist(directory_path: str) -> bool:
```
Comment (tyler-hoffman) [inline great_expectations/data_context/store/checkpoint_store.py:100]: Do we want to bring this over to v1? Seems like the store shouldn't know about directory paths, but I get that this is just cut/pasted.

File: tests/datasource/fluent/integration/test_sql_datasources.py [hunk 36]
```diff
@@ -162,13 +159,7 @@ def _get_exception_details(
...
             dict,
         ]
-    ] = next(  # type: ignore[index, assignment]
-        iter(result.to_json_dict()["run_results"].values())  # type: ignore[call-overload,union-attr]
-    )[
-        "validation_result"  # type: ignore[index]
-    ][
-        "results"  # type: ignore[index]
-    ]
+    ] = next(iter(result.dict()["run_results"].values()))["validation_result"]["results"]
```
Comment (tyler-hoffman) [inline tests/datasource/fluent/integration/test_sql_datasources.py:162]: goodness sakes

**Review by tyler-hoffman — APPROVED** [review]

File: great_expectations/data_context/store/checkpoint_store.py (hunk 35 above)
Comment (cdkini) [reply great_expectations/data_context/store/checkpoint_store.py:100]: Hmm I don't love it but moving this over and changing the name from `V1CheckpointStore` to `CheckpointStore` allows everything to continue functioning like before
